### ISSUE TITLE
Proposal : `buildkit-tekton` local executor project

### ISSUE THREAD

**Issue by vdemeester** [tektoncd/community#674]
TL;DR: adoption of [`buildkit-tekton`](https://github.com/vdemeester/buildkit-tekton) in the `tektoncd` org.

The [`buildkit-tekton`](https://github.com/vdemeester/buildkit-tekton) propose an alternative implementation of the Tekton API, dedicate to local execution. It is currently composed of *two artifcats*:
- a [`buildkit`](https://github.com/moby/buildkit) frontend
- a `tkn-local` command to easily consume this frontend from the comand-line

A bit more context around this can be visible in https://github.com/tektoncd/community/issues/145 and in [CI/CD and the development workflow](https://docs.google.com/document/d/1730VRKxKbvpR9LrmUHs6wjukvkiuL_CDidgf6zDJ7d0/edit#).

Since the inception of the Tekton project, we always focused our work around kubernetes, *but* with a higher end-goal which is, to *probably* in a long-term future, be able to execute Tekton `Task` and `Pipeline` elsewhere. We are trying our hardest in `tektoncd/pipeline` to leak as little as possible from k8s in our APIs.
This alternate implementation / executor is inline with that approach, as it provides an alternative execution flow, and should help us detect where we are leaking too much from Kubernetes.

## Who will own it

- @vdemeester 
- *[any other takers]*

**Comment by jerop** [comment]
discussed this topic at the governance/community working group and agreed that we're interested in adopting buildkit-tekton, but we need to identify the work that needs to be done before adoption (e.g. alignment with design principles)

**Comment by vdemeester** [comment]
@aaron-prindle can you go into more detail on what you mean by a "frontend" over dagger ? I am assuming it would be something that would take a set of Tekton Resources and translate it to the graphql that dagger uses so it can be used by it, am I right ? (If that's the case, it's very similar to what buildkit-tekton aims to do — but one level lower — and, yeah that's something that might be worth to explore, this was in my "personal backlog" to try that out someday 😬 )

**Comment by aaron-prindle** [comment]
@vdemeester exactly, the idea would be translate a set of Tekton Resources -> dagger graph.  This video shows something similar to the concept only for a Github Actions frontend for dagger:
https://www.youtube.com/watch?v=rHk3spfCHFQ

The gist being that instead of using buildkit directly and then writing the additional logic to map artifacts, params, etc. across different Tasks we can likely leverage dagger which has these type of primitives for DAGs on top of buildkit already.

**Comment by vdemeester** [comment]
> The gist being that instead of using buildkit directly and then writing the additional logic to map/pass artifacts, params, etc. across different Tasks we can likely leverage dagger which has these type of primitives for DAGs on top of buildkit already.

Having worked on both, it's a bit higher level yes, but in both case primitives are there. A very thin 1h "test" of this (from last august) is [there](https://gist.github.com/vdemeester/af17928fe6e8fc7593d445566bf948b9). I was essentially slightly simpler than with `buildkit` but not that much different.
